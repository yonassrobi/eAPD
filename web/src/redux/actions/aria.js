export const ARIA_ANNOUNCE_CHANGE = 'ARIA_ANNOUNCE_CHANGE';

export const APD_CREATE_SUCCESS_MSG =
  'You have successfully created an APD. Close this message and continue to fill out the rest of the APD.';

export const ariaAnnounceFFPQuarterly =
  (activityId, year, q, name) => (dispatch, getState) => {
    dispatch({
      type: ARIA_ANNOUNCE_CHANGE,
      message: `${
        getState().budget.activities[activityId].quarterlyFFP.years[year][q][
          name
        ].dollars
      } dollars`
    });
  };

export const ariaAnnounceApdCreateSuccess = () => ({
  type: ARIA_ANNOUNCE_CHANGE,
  message: APD_CREATE_SUCCESS_MSG
});

export const ariaAnnounceApdLoading = () => ({
  type: ARIA_ANNOUNCE_CHANGE,
  message: 'Your APD is loading.'
});

export const ariaAnnounceApdLoaded = () => ({
  type: ARIA_ANNOUNCE_CHANGE,
  message:
    'Your APD is loaded and ready to edit. Changes to this APD will be saved automatically.'
});

export const ariaAnnounceApdLoadingFailure = error => ({
  type: ARIA_ANNOUNCE_CHANGE,
  message: error
});
