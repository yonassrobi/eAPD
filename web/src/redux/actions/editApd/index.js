export * from './symbols';

export * from './assurancesAndCompliance';
export * from './incentivePayments';
export * from './keyPersonnel';
export * from './overview';
export * from './previousActivities';
export * from './stateProfile';
export * from './uploadFile';
export * from './securityPlanning';
