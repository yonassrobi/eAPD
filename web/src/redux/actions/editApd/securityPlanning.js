import { EDIT_APD } from './symbols';

/**
 * Set a summary for Security Interface Plan
 * @param {String} text New summary text
 */
export const setSecurityInterfacePlan = text => ({
  type: EDIT_APD,
  path: '/securityPlanning/securityAndInterfacePlan',
  value: text
});

/**
 * Set a summary for Business Continuity and Disaster Recovery Plan
 * @param {String} text New summary text
 */
export const setBCDRPlan = text => ({
  type: EDIT_APD,
  path: '/securityPlanning/businessContinuityAndDisasterRecovery',
  value: text
});
