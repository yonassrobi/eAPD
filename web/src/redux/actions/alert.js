import { APD_CREATE_SUCCESS_MSG } from './aria';

export const ALERT_SUCCESS = 'ALERT_SUCCESS';
export const ALERT_RESOLVE = 'ALERT_RESOLVE';

export const alertApdCreateSuccess = id => ({
  type: ALERT_SUCCESS,
  apdId: id,
  message: APD_CREATE_SUCCESS_MSG
});

export const resolveAlertMessage = i => dispatch => {
  dispatch({
    type: ALERT_RESOLVE,
    message_index: i
  });
};
