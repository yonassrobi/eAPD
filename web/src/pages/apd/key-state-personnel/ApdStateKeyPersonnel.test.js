import { shallow } from 'enzyme';
import React from 'react';

import {
  plain as KeyPersonnel,
  mapStateToProps,
  mapDispatchToProps
} from './ApdStateKeyPersonnel';
import { removeKeyPersonnel } from '../../../redux/actions/editApd';

describe('apd state profile, Medicaid office component', () => {
  const props = {
    list: [
      {
        key: 'person1',
        name: 'person name',
        email: 'person1@theplace.gov',
        position: 'unobservable',
        fte: { 1: 0.27, 2: 0.27 },
        hasCosts: true,
        costs: { 1: 100, 2: 200 }
      },
      {
        key: 'person2',
        name: '',
        email: 'person2@theplace.gov',
        position: '',
        fte: { 1: 0.72, 2: 0.72 },
        hasCosts: false,
        costs: { 1: 0, 2: 0 }
      }
    ],
    remove: jest.fn(),
    years: ['1', '2']
  };

  const component = shallow(<KeyPersonnel {...props} />);

  beforeEach(() => {
    props.remove.mockClear();
  });

  test('renders correctly', () => {
    expect(component).toMatchSnapshot();
  });

  describe('events', () => {
    const list = component.find('FormAndReviewList');

    it('handles deleting a key person', () => {
      list.prop('onDeleteClick')(0);
      expect(props.remove).toHaveBeenCalledWith(0);
    });
  });

  it('maps state to props', () => {
    expect(
      mapStateToProps({
        apd: {
          data: {
            keyStatePersonnel: {
              keyPersonnel: 'these are my people'
            },
            years: 'and these are my years'
          }
        }
      })
    ).toEqual({
      list: 'these are my people',
      years: 'and these are my years',
      apdType: 'HITECH'
    });
  });

  it('maps dispatch to props', () => {
    expect(mapDispatchToProps).toEqual({
      remove: removeKeyPersonnel
    });
  });
});
