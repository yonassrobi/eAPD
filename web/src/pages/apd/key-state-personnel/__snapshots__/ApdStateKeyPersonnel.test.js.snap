// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`apd state profile, Medicaid office component renders correctly 1`] = `
<FormAndReviewList
  addButtonText="Add Key Personnel"
  allowDeleteAll={false}
  className={null}
  collapsed={[Function]}
  errorCheck={false}
  expanded={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "WrappedComponent": Object {
        "$$typeof": Symbol(react.forward_ref),
        "propTypes": Object {
          "apdType": [Function],
          "index": [Function],
          "item": [Function],
          "savePerson": [Function],
          "setFormValid": [Function],
          "years": [Function],
        },
        "render": [Function],
      },
      "render": [Function],
    }
  }
  list={
    Array [
      Object {
        "costs": Object {
          "1": 100,
          "2": 200,
        },
        "email": "person1@theplace.gov",
        "fte": Object {
          "1": 0.27,
          "2": 0.27,
        },
        "hasCosts": true,
        "key": "person1",
        "name": "person name",
        "position": "unobservable",
      },
      Object {
        "costs": Object {
          "1": 0,
          "2": 0,
        },
        "email": "person2@theplace.gov",
        "fte": Object {
          "1": 0.72,
          "2": 0.72,
        },
        "hasCosts": false,
        "key": "person2",
        "name": "",
        "position": "",
      },
    ]
  }
  noDataMessage="Primary Point of Contact has not been added for this activity."
  onAddClick={[Function]}
  onCancelClick={[Function]}
  onDeleteClick={[Function]}
  years={
    Array [
      "1",
      "2",
    ]
  }
/>
`;
